clients/src/main/java/org/apache/kafka/common/record/RaftUtils.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.record;
+
+import org.apache.kafka.common.message.LeaderChangeMessageData;
+import org.apache.kafka.common.protocol.types.Struct;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Utility class for easy interaction with Raft library.
+ */
+public class RaftUtils {
```

Comment:
Oh, I see. I thought this was in the `raft` package, so I wondered why not use `RaftUtil`. Trying to think of another home... One idea is to expose `MemoryRecordsBuilder.appendControlRecord` so that we can move this into the `raft` package. It would be kind of nice not to clutter the clients library with unnecessary stuff. What do you think?

Reply:
I guess there is already an example with TxnMarker who is also utilizing this client library, so maybe we are good for now just to reside here?

Reply:
I think I'm still inclined to find a new home for these two functions. At a minimum, it's annoying to have both `RaftUtil` and `RaftUtils` in the codebase. Maybe it would be more justifiable as `ControlRecordUtil`?

Reply:
That sounds reasonable to me.